src/server/helper/kmsClientHelper.js

```diff
@@ -0,0 +1,91 @@
+import NyplDataApiClient from '@nypl/nypl-data-api-client';
+import aws from 'aws-sdk';
+
+const kmsClientHelper = (options) => {
+  const {
+    kmsEnvironment,
+    apiBase,
+    tokenUrl,
+    clientId,
+    clientSecret,
+  } = options;
+  const keys = [clientId, clientSecret];
+  const CACHE = {};
+  let decryptKMS;
+  let kms;
+
+  // If we want to use encrypted client id and secret strings, then we need to set up
+  // AWS and the KMS decryption function.
+  if (kmsEnvironment === 'encrypted') {
+    kms = new aws.KMS({
+      region: 'us-east-1',
+    });
+
+    decryptKMS = (key) => {
+      const params = {
+        CiphertextBlob: new Buffer(key, 'base64'),
+      };
+
+      return new Promise((resolve, reject) => {
+        kms.decrypt(params, (err, data) => {
```

Comment:
So overall this module is a factory for making instances of NyplDataApiClient. One of the things it has to do is decrypt creds. Does the app not need to `kms.decrypt` anything else? If this is the only place you need to call `kms.decrypt`, this is fine, but maybe that could be a separate helper module like `kmsDecryptHelper` solely responsible for returning a promise that resolves decrypted text

Reply:
Very good point. Right now only the client id and secret will need to be decrypted but it'd be good to separate it out.